com.unity.render-pipelines.core/Tests/Runtime/CollectionExtensionTests.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnityEngine.Rendering.Tests
+{
+    public class CollectionExtensionTests
+    {
+        static TestCaseData[] s_ListTestsCaseDatas =
+        {
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 1, 2).SetName("Remove middle"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 2).SetName("Remove front"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 6).SetName("Remove all"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 1).SetName("Remove back"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, -1).SetName("Count negative"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, -1, 2).SetName("Index negative"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 0).SetName("Count 0"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 0).SetName("Index 0"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 5).SetName("Count exceeds list size"),
+        };
+
+        [Test, TestCaseSource(nameof(s_ListTestsCaseDatas))]
+        public void CheckRemoveRange(List<int> list, int startIndex, int count)
+        {
+            using (ListPool<int>.Get(out var copy))
+            using (GenericPool<SimpleList>.Get(out var simpleList))
+            {
+                copy.AddRange(list);
+                simpleList.AddRange(list);
+
+                Type exceptionType = null;
+
+                try
+                {
+                    (list as IList<int>).RemoveRange(startIndex, count);
```

Comment:
Test the `Try` API instead, that is the deepest API, this one indeeds is based on a datasource.

The direct API test should only test argument checking & throwing, so it can be simpler and not rely on a datasource